File: clients/pub.py
import httplib, urllib, random

import logging

try:
    import json
except ImportError:
    raise ImportError, "Can't load a json library"

class Pubdis:
    pubdis_nodes = ['127.0.0.1:8000']

    @staticmethod
    def publish(id, action, data):

        domain = random.choice(Pubdis.pubdis_nodes)
        path = '/'+id+'/'+action
        params = urllib.urlencode({'data': json.dumps(data) })

        conn = httplib.HTTPConnection(domain)
        conn.request("POST", path, params)

        conn.sock.settimeout(0.5)

        response = conn.getresponse()

        data = response.read()
        conn.close()

File: clients/pub.js
exports.publish = (function() {

    var http = require('http'),
        pubdis_nodes = ['127.0.0.1:8000'];

    pubdis_nodes.get_random = function() {
        return this[Math.random()*this.length|0];
    }

    return function(id, action, data) {
        var node = pubdis_nodes.get_random().split(':'),
            domain = node[0],
            port = node[1] || 80,
            path = '/'+id+'/'+action;

        var post = http.request({
            host: domain,
            port: port,
            path: path,
            method: 'POST'
        });
        post.on('error',function(e) { console.error("Problem publishing to Pubdis! "+e); });
        post.end('data='+encodeURIComponent(JSON.stringify(data)),'utf8');
    };

})();